### ISSUE TITLE
"Message too long" on FreeBSD

### ISSUE THREAD

**Issue by avkarenow** [nginx/unit#167]
Hello,

I tried to apply a test config with 100 random PHP applications and 100 listeners (config attached).

curl -X PUT -d @php.test --unix-socket /var/run/unit/control.unit.sock http://localhost/config/
{
	"success": "Reconfiguration done."
}

In error log:
2018/09/17 22:20:23 [alert] 44077#102354 sendmsg(7, -1, 2) failed (40: Message too long)
2018/09/17 22:20:23 [info] 44077#102354 *5 shutdown(9, 2) failed (57: Socket is not connected)

Applications work fine but after unitd restart config is empty and I need to recreate it.

System:
FreeBSD 11.2-RELEASE-p2
Unit: unit-1.3_1 from ports.

[php.test.txt](https://github.com/nginx/unit/files/2390148/php.test.txt)

**Comment by avkarenow** [comment]
unit-1.7: the same problem. Any chances for the fix?

2019/01/16 01:29:54 [alert] 91544#100391 sendmsg(7, -1, 2) failed (40: Message too long)

**Comment by VBart** [comment]
@avkarenow Please try to adjust `net.inet.raw.maxdgram` and `net.inet.raw.recvspace` sysctl parameters in your system to 65536.

**Comment by avkarenow** [comment]
Unfortunately, it doesn't change anything.

# sysctl net.local.dgram.maxdgram net.local.dgram.recvspace
net.local.dgram.maxdgram: 65536
net.local.dgram.recvspace: 65536

2019/01/18 22:13:22 [alert] 98466#101777 sendmsg(7, -1, 2) failed (40: Message too long)

**Comment by VBart** [comment]
@avkarenow Have you restarted the Unit daemon after the change?

**Comment by VBart** [comment]
@avkarenow please try the following patch:
```diff
diff -r edd94bc39e80 src/nxt_port_socket.c
--- a/src/nxt_port_socket.c     Wed Jan 09 18:03:48 2019 +0300
+++ b/src/nxt_port_socket.c     Tue Jan 22 19:47:23 2019 +0300
@@ -48,7 +48,7 @@ nxt_port_socket_init(nxt_task_t *task, n
     }
 
     if (max_size == 0) {
-        max_size = 16 * 1024;
+        max_size = 2 * 1024;
     }
 
     if ((size_t) sndbuf < max_size) {
```

**Comment by avkarenow** [comment]
Seems to patch is fixing this problem.

Thanks!

**Comment by avkarenow** [comment]
This patch will be part of next release of Unit?

**Comment by VBart** [comment]
@avkarenow Unfortunately it's not a part of Unit 1.8. This particular change is more like a workaround that aren't good and doesn't guarantee anything. We are working on a better solution, that will be implemented in the next release (Unit 1.9). Thanks for your patience.
